core/services/workflows/syncer/v2/workflow_registry.go

```diff
@@ -497,6 +497,11 @@ func (w *workflowRegistry) syncUsingReconciliationStrategy(ctx context.Context,
 			w.lggr.Debug("shutting down readRegistryStateLoop")
 			return
 		case <-ticker:
+			don, err := w.workflowDonNotifier.WaitForDon(ctx)
+			if err != nil {
+				w.lggr.Errorw("failed to get get don from notifier", "err", err)
+				return
```

Comment:
nit: might be safer to continue here instead of return although right now it will work because the only reason to get an error is via context cancellation.